src/ImageCore.jl

```diff
@@ -33,13 +33,10 @@ const RRArray{To,From,N,M,P} = Base.ReshapedArray{To,N,Base.ReinterpretArray{To,
 const RGArray = Union{Base.ReinterpretArray{<:AbstractGray,M,<:Number,P}, Base.ReinterpretArray{<:Number,M,<:AbstractGray,P}} where {M,P}
 
 # delibrately not export these constants to enable extensibility for downstream packages
-const NumberLike{T<:Number} = Union{T,AbstractGray{T}}
-const RealLike{T<:Real} = NumberLike{T}
-const FloatLike{T<:AbstractFloat} = RealLike{T}
-const FractionalLike{T<:Union{FixedPoint, AbstractFloat}} = RealLike{T}
-const GrayLike{T<:Union{Bool, FixedPoint, AbstractFloat}} = RealLike{T}
-const GenericGrayImage{N, T<:GrayLike} = AbstractArray{<:GrayLike{T}, N}
-const Gray2dImage{T<:GrayLike} = GenericGrayImage{2, T}
+const NumberLike = Union{<:Number,<:AbstractGray}
+const Pixel = Union{<:Number,<:Colorant}
+const GenericGrayImage{T<:NumberLike,N} = AbstractArray{T,N}
```

Comment:
One more potential issue about this is the difference between the following two usages:
```julia
whatis(::GenericImage{T, 2}) where T = "Generic2dImage" # wrong annotation
whatis(::GenericImage{<:Pixel, 2})  = "Generic2dImage" # correct
```

But I guess most developers are able to distinguish them.

Reply:
The first is https://github.com/JuliaLang/julia/issues/6383